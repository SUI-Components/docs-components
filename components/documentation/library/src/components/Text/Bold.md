# Bold
The HTML Bring Attention To element (`b`) is used to draw the reader&#39;s attention to the element&#39;s contents, which are not otherwise granted special importance.

## Installation
`npm install @s-ui/documentation-library`

## Usage

```js
// import Bold DOC component
import { Bold } from '@s-ui/documentation-library/lib/components/Text/Text.js'
```

or

```js
// import all DOC components you need
import { Bold } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
