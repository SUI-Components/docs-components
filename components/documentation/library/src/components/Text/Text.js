/* eslint react/prop-types: 0 */
import React from 'react'
import cx from 'classnames'

import Base from '../Base.core'

export const TEXT_ELEMENT = {
  b: {
    elementType: 'b',
    className: cn => cx('sui-studio-doc-text', 'sui-studio-doc-text-bold', cn)
  },
  strong: {
    elementType: 'strong',
    className: cn => cx('sui-studio-doc-text', 'sui-studio-doc-text-strong', cn)
  },
  i: {
    elementType: 'i',
    className: cn => cx('sui-studio-doc-text', 'sui-studio-doc-text-italic', cn)
  },
  em: {
    elementType: 'em',
    className: cn =>
      cx('sui-studio-doc-text', 'sui-studio-doc-text-emphasized', cn)
  },
  mark: {
    elementType: 'mark',
    className: cn => cx('sui-studio-doc-text', 'sui-studio-doc-text-marked', cn)
  },
  small: {
    elementType: 'small',
    className: cn => cx('sui-studio-doc-text', 'sui-studio-doc-text-small', cn)
  },
  del: {
    elementType: 'del',
    className: cn =>
      cx('sui-studio-doc-text', 'sui-studio-doc-text-deleted', cn)
  },
  ins: {
    elementType: 'ins',
    className: cn =>
      cx('sui-studio-doc-text', 'sui-studio-doc-text-inserted', cn)
  },
  sub: {
    elementType: 'sub',
    className: cn =>
      cx('sui-studio-doc-text', 'sui-studio-doc-text-subscript', cn)
  },
  sup: {
    elementType: 'sup',
    className: cn =>
      cx('sui-studio-doc-text', 'sui-studio-doc-text-superscript', cn)
  }
}

/**
 * Simple plain text styled
 */
const Text = ({elementType = 'span', children, ...props}) => (
  <Base {...props} elementType={elementType}>
    {children}
  </Base>
)

Text.displayName = 'Text'
Text.propTypes = {}
Text.defaultProps = {}

/**
 * The HTML Bring Attention To element (`b`) is used to draw the reader's attention to the element's contents, which are not otherwise granted special importance.
 */
const Bold = ({
  children,
  className,
  elementType = TEXT_ELEMENT.b.elementType,
  ...props
}) => (
  <Text
    {...props}
    className={TEXT_ELEMENT.b.className(className)}
    elementType={elementType}
  >
    {children}
  </Text>
)
Bold.displayName = 'Bold'
Bold.propTypes = {}
Bold.defaultProps = {}

/**
 * The HTML Strong Importance Element (`strong`) indicates that its contents have strong importance, seriousness, or urgency. Browsers typically render the contents in bold type.
 */
const Strong = ({
  children,
  className,
  elementType = TEXT_ELEMENT.strong.elementType,
  ...props
}) => (
  <Text
    {...props}
    className={TEXT_ELEMENT.strong.className(className)}
    elementType={elementType}
  >
    {children}
  </Text>
)
Strong.displayName = 'Strong'
Strong.propTypes = {}
Strong.defaultProps = {}

/**
 * The HTML Idiomatic Text element (`i`) represents a range of text that is set off from the normal text for some reason, such as idiomatic text, technical terms, taxonomical designations, among others.
 */
const Idiomatic = ({
  children,
  className,
  elementType = TEXT_ELEMENT.i.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.i.className(className)}
  >
    {children}
  </Text>
)
Idiomatic.displayName = 'Idiomatic'
Idiomatic.propTypes = {}
Idiomatic.defaultProps = {}

/**
 * The HTML `em` element marks text that has stress emphasis. The <em> element can be nested, with each level of nesting indicating a greater degree of emphasis.
 */
const Emphasis = ({
  children,
  className,
  elementType = TEXT_ELEMENT.em.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.em.className(className)}
  >
    {children}
  </Text>
)
Emphasis.displayName = 'Emphasis'
Emphasis.propTypes = {}
Emphasis.defaultProps = {}

/**
 * The HTML Mark Text element (`mark`) represents text which is marked or highlighted for reference or notation purposes, due to the marked passage's relevance or importance in the enclosing context.
 */
const Mark = ({
  children,
  className,
  elementType = TEXT_ELEMENT.mark.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.mark.className(className)}
  >
    {children}
  </Text>
)
Mark.displayName = 'Mark'
Mark.propTypes = {}
Mark.defaultProps = {}

/**
 * The HTML `small element represents side-comments and small print, like copyright and legal text, independent of its styled presentation. By default, it renders text within it one font-size smaller, such as from small to x-small
 */
const Small = ({
  children,
  className,
  elementType = TEXT_ELEMENT.small.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.small.className(className)}
  >
    {children}
  </Text>
)
Small.displayName = 'Small'
Small.propTypes = {}
Small.defaultProps = {}

/**
 * 	The HTML `del` element represents a range of text that has been deleted from a document.
 */
const Deleted = ({
  children,
  className,
  elementType = TEXT_ELEMENT.del.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.del.className(className)}
  >
    {children}
  </Text>
)
Deleted.displayName = 'Deleted'
Deleted.propTypes = {}
Deleted.defaultProps = {}

/**
 * 	The HTML `ins` element represents a range of text that has been added to a document.
 */
const Inserted = ({
  children,
  className,
  elementType = TEXT_ELEMENT.ins.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.ins.className(className)}
  >
    {children}
  </Text>
)
Inserted.displayName = 'Inserted'
Inserted.propTypes = {}
Inserted.defaultProps = {}

/**
 * The HTML Subscript element (`sub`) specifies inline text which should be displayed as subscript for solely typographical reasons.
 */
const Subscript = ({
  children,
  className,
  elementType = TEXT_ELEMENT.sub.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.sub.className(className)}
  >
    {children}
  </Text>
)
Subscript.displayName = 'Subscript'
Subscript.propTypes = {}
Subscript.defaultProps = {}

/**
 * The HTML Superscript element (`sup`) specifies inline text which is to be displayed as superscript for solely typographical reasons.
 */
const Superscript = ({
  children,
  className,
  elementType = TEXT_ELEMENT.sup.elementType,
  ...props
}) => (
  <Text
    {...props}
    elementType={elementType}
    className={TEXT_ELEMENT.sup.className(className)}
  >
    {children}
  </Text>
)
Superscript.displayName = 'Superscript'
Superscript.propTypes = {}
Superscript.defaultProps = {}

export default Text

export {
  Text,
  Bold,
  Strong,
  Idiomatic,
  Emphasis,
  Mark,
  Small,
  Deleted,
  Inserted,
  Subscript,
  Superscript
}
