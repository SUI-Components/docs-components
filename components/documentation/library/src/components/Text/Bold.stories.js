import React from 'react'

import {Bold as BoldText, TEXT_ELEMENT} from './Text'

export default {
  title: 'Components/Text/Bold',
  component: BoldText,
  argTypes: {
    mode: {
      control: {
        type: 'select',
        options: {
          undefined: undefined,
          light: 'light',
          dark: 'dark'
        }
      }
    }
  }
}

export const Default = props => <BoldText {...props} />

Default.title = 'Components/Text/Bold'

Default.args = {
  children: 'Bold',
  elementType: TEXT_ELEMENT.b.elementType
}
