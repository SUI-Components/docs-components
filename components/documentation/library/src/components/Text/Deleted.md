# Deleted
The HTML `del` element represents a range of text that has been deleted from a document.

## Installation
`npm install @s-ui/documentation-library`

## Usage

```js
// import Deleted DOC component
import { Deleted } from '@s-ui/documentation-library/lib/components/Text/Text.js'
```

or

```js
// import all DOC components you need
import { Deleted } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
