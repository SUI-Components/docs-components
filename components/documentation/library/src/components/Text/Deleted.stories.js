import React from 'react'

import {Deleted as DeletedText, TEXT_ELEMENT} from './Text'

export default {
  title: 'Components/Text/Deleted',
  component: DeletedText,
  argTypes: {
    mode: {
      control: {
        type: 'select',
        options: {
          undefined: undefined,
          light: 'light',
          dark: 'dark'
        }
      }
    }
  }
}

export const Default = props => <DeletedText {...props} />

Default.title = 'Components/Text/Deleted'

Default.args = {
  children: 'Deleted',
  elementType: TEXT_ELEMENT.del.elementType
}
