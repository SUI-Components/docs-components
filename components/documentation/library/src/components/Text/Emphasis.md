# Emphasis
> version undefined

The HTML `em` element marks text that has stress emphasis. The &lt;em&gt; element can be nested, with each level of nesting indicating a greater degree of emphasis.

## Installation
`npm install @s-ui/documentation-library`

## Usage
```scss
// include Emphasis SASS module
@include '@s-ui/documentation-library/lib/components/Text/Text.scss'
```

```js
// import Emphasis DOC component
import { Emphasis } from '@s-ui/documentation-library/lib/components/Text/Text.js'
```

or

```scss
// include all SASS modules
@include '@s-ui/documentation-library/lib/index.scss'
```

```js
// import all DOC components you need
import { Emphasis } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
