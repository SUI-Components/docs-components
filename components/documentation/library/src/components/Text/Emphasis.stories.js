import React from 'react'

import {Emphasis as EmphasisText, TEXT_ELEMENT} from './Text'

export default {
  title: 'Components/Text/Emphasis',
  component: EmphasisText,
  argTypes: {
    mode: {
      control: {
        type: 'select',
        options: {
          undefined: undefined,
          light: 'light',
          dark: 'dark'
        }
      }
    }
  }
}

export const Default = props => <EmphasisText {...props} />

Default.title = 'Components/Text/Emphasis'

Default.args = {
  children: 'Emphasis',
  elementType: TEXT_ELEMENT.em.elementType
}
