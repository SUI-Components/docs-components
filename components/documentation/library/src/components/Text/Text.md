# Text
> version undefined

Simple plain text styled

## Installation
`npm install @s-ui/documentation-library`

## Usage
```scss
// include Text SASS module
@include '@s-ui/documentation-library/lib/components/Text/Text.scss'
```

```js
// import Text DOC component
import { Text } from '@s-ui/documentation-library/lib/components/Text/Text.js'
```

or

```scss
// include all SASS modules
@include '@s-ui/documentation-library/lib/index.scss'
```

```js
// import all DOC components you need
import { Text } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
