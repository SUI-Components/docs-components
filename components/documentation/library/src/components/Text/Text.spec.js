/* global expect */
import React, {createRef} from 'react'
import ReactDOM from 'react-dom'

import {Text} from './Text'

import setupEnvironment from '../../helpers/setupEnvironment'

describe('components', () => {
  describe('Text', () => {
    describe('default', () => {
      const Component = Text
      const setup = setupEnvironment(Component)

      test('should render without crashing', () => {
        // Given
        const props = {}

        // When
        const component = <Component {...props} />

        // Then
        const div = document.createElement('div')
        ReactDOM.render(component, div)
        ReactDOM.unmountComponentAtNode(div)
      })

      test('should NOT render null', () => {
        // Given
        const props = {}

        // When
        const {container} = setup(props)

        // Then
        expect(container.innerHTML).toBeString()
        expect(container.innerHTML).not.toHaveLength(0)
      })

      test('should have propTypes and defaultProps ', () => {
        // Given
        const component = Component

        // When

        // Then
        expect(component.hasOwnProperty('propTypes')).toBeTruthy()
        expect(component.hasOwnProperty('defaultProps')).toBeTruthy()
      })

      test('should have proper displayName attribute', () => {
        // Given
        // const props = {}

        // When
        const component = Component

        // Then
        expect(component.displayName).toBeString()
        expect(component.displayName).not.toHaveLength(0)
        expect(component.displayName).toMatchSnapshot()
      })

      describe('forwardRef', () => {
        test('should have reference', () => {
          // Given
          const props = {}
          const ref = createRef()

          // When
          const component = <Component {...props} ref={ref} />
          const div = document.createElement('div')
          ReactDOM.render(component, div)

          // Then
          expect(ref.current).not.toBe(undefined)
          expect(ref.current.nodeName).toBe('SPAN')
        })

        test('should have defined elementType reference', () => {
          // Given
          const props = {elementType: 'div'}
          const ref = createRef()

          // When
          const component = <Component {...props} ref={ref} />
          const div = document.createElement('div')
          ReactDOM.render(component, div)

          // Then
          expect(ref.current).not.toBe(undefined)
          expect(ref.current.nodeName).toBe('DIV')
        })
      })
    })
  })
})
