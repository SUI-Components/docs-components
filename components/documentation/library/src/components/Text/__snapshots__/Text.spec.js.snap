// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Text default should have proper displayName attribute 1`] = `"Text"`;
